lib/World.luau

```diff
@@ -259,40 +413,7 @@ end
 	@param ... ComponentInstance -- The component values to spawn the entity with.
 ]=]
 function World:replace(id, ...)
-	if not self:contains(id) then
-		error(ERROR_NO_ENTITY, 2)
-	end
-
-	local components = {}
-	local metatables = {}
-	local entity = self:_getEntity(id)
-
-	for i = 1, select("#", ...) do
-		local newComponent = select(i, ...)
-
-		assertValidComponentInstance(newComponent, i)
-
-		local metatable = getmetatable(newComponent)
-
-		if components[metatable] then
-			error(("Duplicate component type at index %d"):format(i), 2)
-		end
-
-		self:_trackChanged(metatable, id, entity[metatable], newComponent)
-
-		components[metatable] = newComponent
-		table.insert(metatables, metatable)
-	end
-
-	for metatable, component in pairs(entity) do
-		if not components[metatable] then
-			self:_trackChanged(metatable, id, component, nil)
-		end
-	end
-
-	self._entityMetatablesCache[id] = metatables
-
-	self:_transitionArchetype(id, components)
+	bufferCommand(self, { type = "replace", entityId = id, components = { ... } })
```

Comment:
You should never have to buffer replace as it doesn't make any structural changes, this would just result in extra overhead

Reply:
Replace does make structural changes.